korio/src/commonMain/kotlin/korlibs/io/serialization/xml/Xml.kt

```diff
@@ -253,7 +253,15 @@ data class Xml(
 				val value = r.readUntilIncluded(';') ?: ""
 				val full = "&$value"
 				when {
-					value.startsWith('#') -> append(value.substring(1, value.length - 1).toInt().toChar())
+					value.startsWith('#') -> {
```

Comment:
Perhaps this should be in a private method to adhere to the Single Responsibility Principle?